monix-reactive/shared/src/main/scala/monix/reactive/Observable.scala

```diff
@@ -3719,6 +3724,33 @@ abstract class Observable[+A] extends Serializable { self =>
   final def filter(p: A => Boolean): Observable[A] =
     self.liftByOperator(new FilterOperator(p))
 
+  /** Version of [[filter]] that can work with a predicate expressed by
+    * a [[monix.eval.Task]].
+    *
+    * @see [[filterEvalF]] for a version that works with a generic
+    *      `F[_]` (e.g. `cats.effect.IO`, Scala's `Future`),
+    *      powered by [[monix.eval.TaskLike]]
+    */
+  final def filterEval(p: A => Task[Boolean]): Observable[A] =
+    self
+      .mapEval(a ⇒ p(a).map((a, _)))
+      .filter(_._2)
+      .map(_._1)
```

Comment:
You can replace `filter` and `map` with `collect`. It would be slightly more efficient.

`filter(p).map(f) <-> collect { x if p(x) => f(x) }`

Reply:
There no way we're going to do that, since it requires the logic of `mapEval` and `mapEval` is extremely complicated and that logic is already duplicated from `flatMap`.

Reply:
https://github.com/monix/monix/pull/760/commits/0c01301b27b6434ff91a67af2ffe8a795552f7d7